src/main/java/com/marianbastiurea/lifeofbees/Apiary.java

```diff
@@ -45,32 +55,97 @@ public void setHoneys(List<Honey> honeys) {
     public void addHive(Hive hive) {
         hives.add(hive);
     }
+
     // Add a honey to the apiary
     public void addHoney(Honey honey) {
         honeys.add(honey);
     }
 
     @Override
     public String toString() {
-        return "Apiary{" +
-                "hives=" + hives +
-                ", honeys=" + honeys +
+        return "{" +
+                "numberOfHives=" + this.numberOfHives +
+                ", hives=" + this.hives +
+                ", honeys=" + this.honeys +
                 '}';
     }
 
-    private void splitHive(Hive hive) {
-        // Create a new hive with half of the honey frames and half of the eggs frames
-        Hive newHive = new Hive();
-        newHive.setAgeOfQueen(); // Set age of the new queen to 0
-
-        int halfNumberOfHoneyFrame = hive.getNumberOfHoneyFrame() / 2;
-        int halfNumberOfEggsFrame = hive.getNumberOfEggsFrame() / 2;
-
-        newHive.setNumberOfHoneyFrame(halfNumberOfHoneyFrame);
-        newHive.setNumberOfEggsFrame(halfNumberOfEggsFrame);
-
-        // Transfer the new hive to the apiary
-        hives.add(newHive);
+    public void splitHive() {
+
+        /*
+        this method will split a hive in two equal half. First will check boolean itWasSplit. If this is false
+        hive will be split. A hive could be split only once in a year.
+         */
+
+        List<Hive> newHives = new ArrayList<>();
+        for (Hive hive : hives) {
+            if (hive.getNumberOfEggsFrame() == 6&& hive.isItWasSplit()==false) {
+                Hive newHive = new Hive();// changes  Hive new Hive=new Hive
+                newHive.setId(this.getNumberOfHives() + 1);
+                newHive.setItWasSplit(true);
+                newHive.setNumberOfEggsFrame(hive.getNumberOfEggsFrame());
+                newHive.setNumberOfHoneyFrame(hive.getNumberOfHoneyFrame());
+                newHive.setNumberOfBees(hive.getNumberOfBees() / 2);
+                hive.setNumberOfBees(hive.getNumberOfBees()/2);
+                hive.setItWasSplit(true);
+                Queen queen = new Queen();
+                Honey honey=new Honey();
+                newHive.setQueen(queen);
+                newHive.setHoney(honey);
+                newHive.getQueen().setAgeOfQueen(0);
```

Comment:
Use a constructor with parameters to instantiate Queen instead of using setters.